packaging/docker/fdb-aws-s3-credentials-fetcher/fdb-aws-s3-credentials-fetcher.go

```diff
@@ -0,0 +1,151 @@
+    s3Endpoint := fmt.Sprintf("s3.%s.amazonaws.com", region)
+    
+    // Bulkload hostname is like this backup-112664522426-us-west-2.s3-us-west-2.amazonaws.com
+    // i.e. bucket and then s3.region.amazonaws.com.
+    // So add records for this format and for s3.REGION.amazonaws.com too.
+    blobCreds := BlobCredentials{
+        Accounts: map[string]Account{
+            "@" + s3Endpoint: {
+                Secret:  secretKey,
+                APIKey:  accessKey,
+                Token:   token,
+            },
+            "@" + s3Endpoint + ":443": {
+                Secret:  secretKey,
+                APIKey:  accessKey,
+                Token:   token,
+            },
+            "@" + bucket + "." + s3Endpoint: {
+                Secret:  secretKey,
+                APIKey:  accessKey,
+                Token:   token,
+            },
+            "@" + bucket + "." + s3Endpoint + ":443": {
+                Secret:  secretKey,
+                APIKey:  accessKey,
+                Token:   token,
+            },
+        },
+    }
+
+    data, err := json.Marshal(blobCreds)
+    if err != nil {
+        return fmt.Errorf("failed to marshal credentials: %v", err)
+    }
+
+    if err := os.WriteFile(credFile, data, 0644); err != nil {
+        return fmt.Errorf("failed to write credentials file: %v", err)
+    }
+
+    return nil
+}
+
+func refreshCredentials(bucket, region, credFile string) error {
+    ctx := context.Background()
+    
+    // Load the AWS SDK configuration - this will automatically use IRSA when running in EKS
+    cfg, err := config.LoadDefaultConfig(ctx, 
+        config.WithRegion(region),
+    )
+    if err != nil {
+        return fmt.Errorf("unable to load SDK config: %v", err)
+    }
+
+    // Get current credentials
+    creds, err := cfg.Credentials.Retrieve(ctx)
+    if err != nil {
+        return fmt.Errorf("failed to get credentials: %v", err)
+    }
+
+    // Write credentials to file
+    if err := writeCredentialsFile(bucket, region, credFile, creds.AccessKeyID, creds.SecretAccessKey, creds.SessionToken); err != nil {
+        return fmt.Errorf("failed to write credentials: %v", err)
+    }
+
+    return nil
+}
+
+func main() {
+    // Add usage message
+    flag.Usage = func() {
+        fmt.Fprintf(os.Stderr, "Usage: %s [options]\n\n", os.Args[0])
+        fmt.Fprintf(os.Stderr, "A credential fetcher for AWS S3 that continuously refreshes credentials for FDB blob storage.\n")
+        fmt.Fprintf(os.Stderr, "Options:\n")
+        flag.PrintDefaults()
+    }
+
+    defaultRegion := "us-west-2"
+    if envRegion := os.Getenv("AWS_REGION"); envRegion != "" {
+        defaultRegion = envRegion
+    }
+    
+    region := flag.String("region", defaultRegion, "AWS region for S3 endpoint (default from AWS_REGION env var)")
+    dir := flag.String("dir", "", "Directory path where credentials file will be stored")
```

Comment:
We could use the same flag package as we use in the fdb-kubernetes-monitor package if we add this utility to the foundationdb repo: https://github.com/apple/foundationdb/blob/main/fdbkubernetesmonitor/main.go#L119-L140.

Reply:
Used pflag instead of flag.